proposals/poc/witx/proposal_symmetric.md

```diff
@@ -0,0 +1,1418 @@
+
+- <a href="#crypto_errno.unsupported_option" name="crypto_errno.unsupported_option"></a> `unsupported_option`
+The requested option is not supported by the currently selected algorithm.
+
+- <a href="#crypto_errno.invalid_key" name="crypto_errno.invalid_key"></a> `invalid_key`
+An invalid or incompatible key was supplied.
+
+The key may not be valid, or was generated for a different algorithm or parameters set.
+
+- <a href="#crypto_errno.invalid_length" name="crypto_errno.invalid_length"></a> `invalid_length`
+The currently selected algorithm doesn't support the requested output length.
+
+This error is thrown by non-extensible hash functions, when requesting an output size larger than they produce out of a single block.
+
+- <a href="#crypto_errno.verification_failed" name="crypto_errno.verification_failed"></a> `verification_failed`
+A signature or authentication tag verification failed.
+
+- <a href="#crypto_errno.rng_error" name="crypto_errno.rng_error"></a> `rng_error`
+A secure random numbers generator is not available.
+
+The requested operation requires random numbers, but the host cannot securely generate them at the moment.
+
+- <a href="#crypto_errno.algorithm_failure" name="crypto_errno.algorithm_failure"></a> `algorithm_failure`
+An error was returned by the underlying cryptography library.
+
+The host may be running out of memory, parameters may be incompatible with the chosen implementation of an algorithm or another unexpected error may have happened.
+
+Ideally, the specification should provide enough details and guidance to make this error impossible to ever be thrown.
+
+Realistically, the WASI crypto module cannot possibly cover all possible error types implementations can return, especially since some of these may be language-specific.
+This error can thus be thrown when other error types are not suitable, and when the original error comes from the cryptographic primitives themselves and not from the WASI module.
+
+- <a href="#crypto_errno.invalid_signature" name="crypto_errno.invalid_signature"></a> `invalid_signature`
+The supplied signature is invalid, or incompatible with the chosen algorithm.
+
+- <a href="#crypto_errno.closed" name="crypto_errno.closed"></a> `closed`
+An attempt was made to close a handle that was already closed.
+
+- <a href="#crypto_errno.invalid_handle" name="crypto_errno.invalid_handle"></a> `invalid_handle`
+A function was called with an unassigned handle, a closed handle, or handle of an unexpected type.
+
+- <a href="#crypto_errno.overflow" name="crypto_errno.overflow"></a> `overflow`
+The host needs to copy data to a guest-allocated buffer, but that buffer is too small.
+
+- <a href="#crypto_errno.internal_error" name="crypto_errno.internal_error"></a> `internal_error`
+An internal error occurred.
+
+This error is reserved to internal consistency checks, and must only be sent if the internal state of the host remains safe after an inconsistency was detected.
```

Comment:
This would indicate a bug, right? And should probably be `unreachable`?

Reply:
That would absolutely indicate a bug, and should be `unreachable`.

Is there a way to mark an error `unreachable` in a witx` definition?